src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2SecurityTokenHandler.cs

```diff
@@ -500,6 +525,35 @@ internal static bool IsSaml2Assertion(XmlReader reader)
             return reader.IsStartElement(Saml2Constants.Elements.Assertion, Saml2Constants.Namespace);
         }
 
+        /// <summary>
+        /// Indicates if the assertion string represents an EncryptedAssertion.
+        /// </summary>
+        /// <param name="assertion">String representation of an assertion.</param>
+        /// <returns>'true' if assertion is encrypted, 'false' otherwise.</returns>
+        internal static bool IsSaml2EncryptedAssertion(string assertion)
+        {
+            using (var stringReader = new StringReader(assertion))
+            {
+                var settings = new XmlReaderSettings { DtdProcessing = DtdProcessing.Prohibit };
+#if NET45 || NET451
+                settings.XmlResolver = null;
+#endif
+                using (var reader = XmlReader.Create(stringReader, settings))
+                {
+                    try
+                    {
+                        reader.MoveToContent();
+                        return reader.IsStartElement(Saml2Constants.Elements.EncryptedAssertion, Saml2Constants.Namespace);
+                    }
+                    catch (Exception)
```

Comment:
is it possible to catch only specific exceptions? Ideally some system exceptions should not be swallowed.

Reply:
I'll think about this. The goal of this method is just to return True if first/start element of provided string is EncryptedAssertion and return False otherwise. I don't really care at that point if the string is valid XML or not.

However, I think that it definitely adds value to include exception message to log information.

Reply:
addressed with: 6434db7e54ff310012f06c57624bc5eaf5b09237

Reply:
another reason for me asking is that it's a best practice to only catch exceptions you know what are. as mentioned certain system exceptions should ideally never be captured, it's better from a security perspective to let the code fail with an exception.

---
In reply to: [221345997](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1025#discussion_r221345997) [](ancestors = 221345997)